### PR TITLE
Cog refactor

### PR DESCRIPTION
**Pull request by fiskenslakt**
This PR refactors the project into the much cleaner and easier to work with "cogs" structure. Our `__main__.py` entry point now serves to just load any and all cogs in the `/cogs` package, start the bot, and then clean up our db connection when it halts.

I've moved the root logger initialization to `__init__.py` so `__main__.py` and any cogs can correctly create child loggers from it, so now logs will actually show what module they're coming from. I also changed it to a rotating file handler.

I've also moved all global settings and .env loading to a `constants.py` module, using pydantic to load from .env and any environment variables should they exist.

This PR doesn't add any new features, but it does fix a bug I didn't realize was introduced not long ago where it was no longer possible to give points to someone by user mention (@'ing them).

I tested most of the main features and edge cases, but not all of them cuz I'm lazy. Hopefully I didn't break anything.

### CODE REVIEW

File: pzsd_bot/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from pathlib import Path
+
+from pzsd_bot.constants import Bot
+
+logger = logging.getLogger()
+logger.setLevel(Bot.log_level)
+log_file = Path(__file__).parents[1] / Path("logs", "pzsd_bot.log")
+log_file.parent.mkdir(exist_ok=True)
+handler = handlers.RotatingFileHandler(
+    log_file, maxBytes=1024 * 1024 * 3, backupCount=2, encoding="utf8"
```
Comment (sethdandridge) [inline pzsd_bot/__init__.py:12]: prety sure utf 8 is the default


3 * 1024**2 is cuter

File: pzsd_bot/constants.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+from enum import Enum
+
+from pydantic_settings import BaseSettings
+
+
+class EnvConfig(BaseSettings, env_file=".env", extra="ignore"):
+    """Default config to pull from .env file."""
```
Comment (sethdandridge) [inline pzsd_bot/constants.py:7]: are these really constant then tho

**Review by sethdandridge — APPROVED** [review]
lgtmpc

File: pzsd_bot/__init__.py (hunk 1 above)
Comment (fiskenslakt) [reply pzsd_bot/__init__.py:12]: Looks to be `None` actually: https://docs.python.org/3/library/logging.handlers.html#logging.handlers.RotatingFileHandler

You're right on the second point though

File: pzsd_bot/constants.py (hunk 2 above)
Comment (fiskenslakt) [reply pzsd_bot/constants.py:7]: haha, well. True, not everything in it is a constant, I just named the module that. It's a place for constants and settings. Do you think it's confusing or are you just being cheeky?

Comment (sethdandridge) [reply pzsd_bot/constants.py:7]: settings.py

File: pzsd_bot/__init__.py (hunk 1 above)
Comment (sethdandridge) [reply pzsd_bot/__init__.py]: If it's None does it use system default? Has to choose some encoding

File: pzsd_bot/constants.py (hunk 2 above)
Comment (fiskenslakt) [reply pzsd_bot/constants.py:7]: Fair enough, I'll rename

File: pzsd_bot/__init__.py (hunk 1 above)
Comment (fiskenslakt) [reply pzsd_bot/__init__.py]: I had the same thought, but was too lazy to look any further into it. Doesn't hurt to be explicit.

**Pull request merged**